rosbot_xl_description/urdf/rosbot_xl_macro.urdf.xacro

```diff
@@ -87,49 +93,73 @@
+          </sensor>
+        </xacro:if>
       </xacro:if>
     </ros2_control>
 
     <xacro:if value="${use_sim}">
-      <gazebo>
-        <plugin filename="ign_ros2_control-system" name="ign_ros2_control::IgnitionROS2ControlPlugin">
-          <xacro:if value="${mecanum}">
-            <parameters>$(find rosbot_xl_hardware)/config/mecanum_drive_controller.yaml</parameters>
-          </xacro:if>
-          <xacro:unless value="${mecanum}">
-            <parameters>$(find rosbot_xl_hardware)/config/diff_drive_controller.yaml</parameters>
-          </xacro:unless>
-          <controller_manager_prefix_node_name>simulation</controller_manager_prefix_node_name>
-          <ros>
-            <remapping>/rosbot_xl_base_controller/cmd_vel_unstamped:=/cmd_vel</remapping>
-          </ros>
-        </plugin>
-      </gazebo>
-      <gazebo reference="imu_link">
-        <sensor name="imu" type="imu">
-          <plugin filename="ignition-gazebo-imu-system" name="ignition::gazebo::systems::Imu"></plugin>
-          <always_on>true</always_on>
-          <update_rate>25</update_rate>
-          <topic>imu/data_raw</topic>
-          <visualize>false</visualize>
-          <enable_metrics>false</enable_metrics>
-          <frame_id>imu_link</frame_id>
-          <ignition_frame_id>imu_link</ignition_frame_id>
-        </sensor>
-      </gazebo>
-    </xacro:if>
+      <xacro:if value="${simulation_engine == 'gazebo-ignition'}">
+        <gazebo>
+          <plugin filename="ign_ros2_control-system" name="ign_ros2_control::IgnitionROS2ControlPlugin">
+            <xacro:if value="${mecanum}">
+              <parameters>$(find rosbot_xl_hardware)/config/mecanum_drive_controller.yaml</parameters>
+            </xacro:if>
+            <xacro:unless value="${mecanum}">
+              <parameters>$(find rosbot_xl_hardware)/config/diff_drive_controller.yaml</parameters>
+            </xacro:unless>
+            <controller_manager_prefix_node_name>simulation</controller_manager_prefix_node_name>
+            <ros>
+              <remapping>/rosbot_xl_base_controller/cmd_vel_unstamped:=/cmd_vel</remapping>
+            </ros>
+          </plugin>
+        </gazebo>
+        <gazebo reference="imu_link">
+          <sensor name="imu" type="imu">
+            <plugin filename="ignition-gazebo-imu-system" name="ignition::gazebo::systems::Imu"></plugin>
+            <always_on>true</always_on>
+            <update_rate>25</update_rate>
+            <topic>imu/data_raw</topic>
+            <visualize>false</visualize>
+            <enable_metrics>false</enable_metrics>
+            <frame_id>imu_link</frame_id>
+            <ignition_frame_id>imu_link</ignition_frame_id>
+          </sensor>
+        </gazebo>
+      </xacro:if>
 
+      <xacro:if value="${simulation_engine == 'webots'}">
+        <webots>
+          <plugin type="webots_ros2_control::Ros2Control" />
+          <device reference="laser" type="Lidar">
```

Comment:
I moved it directly to the [webots_ros2](https://github.com/husarion/webots_ros2/tree/humble/webots_ros2_husarion/webots_ros2_husarion/urdf) because is not necessary to add `webots` tags to  [ros_components_description](https://github.com/husarion/ros_components_description).

In (https://github.com/husarion/ros_components_description) there are gazebo sensors parameters in urdfs files but for the webots sensors - parameters are inside the .proto files. For example the lidar properties are in [webots](https://github.com/husarion/webots).

Reply:
But is still will be included to `rosbot_xl_macro` - in this urdf only base of the robot should be defined. Lidar is only later included as a component. I think that it would be better to move
```
<device reference="laser" type="Lidar">
        <ros>
          <topicName>/scan</topicName>
          <updateRate>12</updateRate>
        </ros>
      </device>
```
to lidar, and include other webots configs directly in the `rosbot_xl_macro` (not through `rosbot_xl_webots.urdf.xacro`).